### PR TITLE
Feature/cc 0043 google login

### PR DESCRIPTION
**Pull request by Jzeva**
Google Login Feature

### CODE REVIEW

**Review by stevenliu986 — APPROVED** [review]
LGTM

File: src/auth/auth.service.ts [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+      return newUserInfo;
+    }
+    const userInfo = {
+      googleId: sub,
+      email: email,
+      firstName: given_name,
+      lastName: family_name,
+    };
+    // Return the user who has already existed in the database
+    return userInfo;
```
Comment (chui-mel) [inline src/auth/auth.service.ts:61]: extract all user query and create related code to UserService

File: src/auth/auth.service.ts [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    }
+    return user;
+  }
+
+  /**
+   * Create new user
+   * @param {CreateUserDto} createUserDto
+   * @returns {User}
+   */
+  async create(createUserDto: CreateUserDto): Promise<User> {
```
Comment (chui-mel) [inline src/auth/auth.service.ts:91]: all those user related to userservice

File: src/auth/auth.service.ts [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+    // return the user info which has been updated.
+    return updatedUser;
+  }
+
+  /**
+   * Remove user by Id
+   * @param {ObjectId} id
+   * @returns {User}
+   */
+  async removeUserById(id: ObjectId): Promise<{ message?: string }> {
```
Comment (chui-mel) [inline src/auth/auth.service.ts:122]: same here

File: src/auth/auth.service.ts [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+    return user;
+  }
+
+  /**
+   * Update user by ID
+   * @param {CreateUserDto}id
+   * @param updateUserDto
+   * @returns {User}
+   */
+  async updateUserById(id: ObjectId, updateUserDto: UpdateUserDto): Promise<User> {
```
Comment (chui-mel) [inline src/auth/auth.service.ts:102]: same here

File: src/auth/schemas/user.schema.ts [hunk 5]
```diff
@@ -0,0 +1,33 @@
+import { Optional } from "@nestjs/common";
+import { Schema, Prop, SchemaFactory } from "@nestjs/mongoose";
+import { Document } from "mongoose";
+
+@Schema({ timestamps: true })
+export class User extends Document {
```
Comment (chui-mel) [inline src/auth/schemas/user.schema.ts:6]: we need an objectId as primary key

Comment (Jzeva) [reply src/auth/schemas/user.schema.ts:6]: Normally, MongoDB will add a field called "_id" whose type is ObjectID, and this field("_id") will be reserved to be the primary key if we did not set our own id as "_id". I noticed you left same comment at other devs' pull request.  Maybe this is the answer you want to get. Please remind me if I said anything wrong.

Reference link: 
https://www.mongodb.com/docs/manual/core/document/
https://www.mongodb.com/docs/manual/reference/glossary/
image
image

**Review by EmilyGong111 — APPROVED** [review]
Good!

File: src/user/user.service.ts [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+   * Remove user by Id
+   * @param {ObjectId} id
+   * @returns {User}
+   */
+  async removeUserById(id: ObjectId): Promise<{ message?: string }> {
+    const user = await this.userModel.findById(id).exec();
+
+    // if there is no such a user or the user has been deleted, then throw an error.
+    if (!user || user.isDeleted) {
+      throw new NotFoundException("court not found");
```
Comment (GitHubEason) [inline src/user/user.service.ts:74]: do you mean user?

**Pull request merged**